fix: injector module breaks when used a second time.

## Description

Using the Injector Module multiple times in one process causes a crash, this prevents some system test configurations.

## Expected behavior

To be able to create and use multiple containers with the module. 

## Actual behavior

```
=================================================== test session starts ===================================================
platform linux -- Python 3.11.6, pytest-7.4.2, pluggy-1.3.0 -- /VENV/bin/python
cachedir: .pytest_cache
rootdir: /ROOT
configfile: pyproject.toml
plugins: mypy-0.10.3, pylint-0.21.0, cov-4.1.0, pydocstyle-2.3.2, typeguard-4.1.5, anyio-3.7.1, asyncio-0.21.1
asyncio: mode=Mode.STRICT
collected 1 item                                                                                                               

tests/bug.py::test_bug FAILED                                                                                       [100%]

======================================================== FAILURES =========================================================
________________________________________________________ test_bug _________________________________________________________

    def test_bug() -> None:
        r = banshee.Registry()
        c = injector.Injector(banshee.extra.injector.BansheeModule(r))
        c.get(banshee.Bus)
>       c = injector.Injector(banshee.extra.injector.BansheeModule(r))

tests/bug.py:10: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/VENV/lib/python3.11/site-packages/injector/__init__.py:923: in __init__
    self.binder.install(module)
/VENV/lib/python3.11/site-packages/injector/__init__.py:564: in install
    instance(self)
/VENV/lib/python3.11/site-packages/injector/__init__.py:869: in __call__
    self.configure(binder)
/VENV/lib/python3.11/site-packages/banshee/extra/injector.py:95: in configure
    binder.bind(injector.inject(banshee.IdentityMiddleware))
/VENV/lib/python3.11/site-packages/injector/__init__.py:1410: in inject
    inject(cast(Any, constructor_or_class).__init__)
/VENV/lib/python3.11/site-packages/injector/__init__.py:1415: in inject
    read_and_store_bindings(function, bindings)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

f = <slot wrapper '__init__' of 'object' objects>, bindings = {}

    @private
    def read_and_store_bindings(f: Callable, bindings: Dict[str, type]) -> None:
        function_bindings = getattr(f, '__bindings__', None) or {}
        if function_bindings == 'deferred':
            function_bindings = {}
        merged_bindings = dict(function_bindings, **bindings)
    
        if hasattr(f, '__func__'):
            f = cast(Any, f).__func__
>       cast(Any, f).__bindings__ = merged_bindings
E       AttributeError: 'wrapper_descriptor' object has no attribute '__bindings__'

/VENV/lib/python3.11/site-packages/injector/__init__.py:1477: AttributeError
================================================= short test summary info =================================================
FAILED tests/bug.py::test_bug - AttributeError: 'wrapper_descriptor' object has no attribute '__bindings__'
==================================================== 1 failed in 0.06s ====================================================
```

## Possible fixes

The problem seems to stem from calling `injector.inject()` on a class after it has already been used for injection.

This happens in the `configure` method of `BansheeModule`.

The calls seem to be superfluous and can be removed without changing functionality.

## Steps to Reproduce

```py
import banshee
import banshee.extra.injector
import injector


def test_bug() -> None:
    r = banshee.Registry()
    c = injector.Injector(banshee.extra.injector.BansheeModule(r))
    c.get(banshee.Bus)
    c = injector.Injector(banshee.extra.injector.BansheeModule(r))
```

## Your environment

* Version used: 0.1.0

--- a/.dictionary
+++ b/.dictionary
@@ -3,6 +3,7 @@ cls
 composable
 configurationerror
 iterable
+lineno
 mypy
 pragma
 py

--- a/.github/workflows/release-please.yml
+++ b/.github/workflows/release-please.yml
@@ -11,40 +11,40 @@ jobs:
     env:
       PACKAGE_NAME: "banshee"
     steps:
-    - name: release please
-      uses: google-github-actions/release-please-action@v3
-      id: release
-      with:
-        release-type: python
-        package-name: ${{ env.PACKAGE_NAME }}
+      - name: release please
+        uses: google-github-actions/release-please-action@v3
+        id: release
+        with:
+          release-type: python
+          package-name: ${{ env.PACKAGE_NAME }}
 
-    - name: fetch code
-      uses: actions/checkout@v4
-      if: ${{ steps.release.outputs.release_created }}
+      - name: fetch code
+        uses: actions/checkout@v4
+        if: ${{ steps.release.outputs.release_created }}
 
-    - name: set python version
-      if: ${{ steps.release.outputs.release_created }}
-      uses: actions/setup-python@v4
-      with:
-        python-version: "3.10"
+      - name: set python version
+        if: ${{ steps.release.outputs.release_created }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
 
-    - name: install dependencies
-      uses: artisanal-actions/poetry-install@v1
-      if: ${{ steps.release.outputs.release_created }}
+      - name: install dependencies
+        uses: artisanal-actions/poetry-install@v1
+        if: ${{ steps.release.outputs.release_created }}
 
-    - name: build release
-      if: ${{ steps.release.outputs.release_created }}
-      run: poetry build
-      
-    - name: upload assets
-      uses: softprops/action-gh-release@v1
-      if: ${{ steps.release.outputs.release_created }}
-      with:
-        tag_name: ${{ steps.release.outputs.tag_name }}
-        files: dist/*
+      - name: build release
+        if: ${{ steps.release.outputs.release_created }}
+        run: poetry build
 
-    - name: publish release
-      if: ${{ steps.release.outputs.release_created }}
-      run: poetry publish
-      env:
-        POETRY_PYPI_TOKEN_PYPI: ${{ secrets.PYPI_AUTH_TOKEN }}
+      - name: upload assets
+        uses: softprops/action-gh-release@v1
+        if: ${{ steps.release.outputs.release_created }}
+        with:
+          tag_name: ${{ steps.release.outputs.tag_name }}
+          files: dist/*
+
+      - name: publish release
+        if: ${{ steps.release.outputs.release_created }}
+        run: poetry publish
+        env:
+          POETRY_PYPI_TOKEN_PYPI: ${{ secrets.PYPI_AUTH_TOKEN }}

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -3,36 +3,38 @@ name: tests
 on:
   push:
     paths-ignore:
-      - 'docs/**'
+      - "docs/**"
     branches:
       - main
   pull_request:
     paths-ignore:
-      - 'docs/**'
+      - "docs/**"
     branches:
-      - '**'
+      - "**"
 
 jobs:
   tests:
     runs-on: ubuntu-latest
 
     steps:
-    - name: fetch code
-      uses: actions/checkout@v4
+      - name: fetch code
+        uses: actions/checkout@v4
 
-    - name: install en-gb dictionary
-      run: sudo apt-get install -y hunspell-en-gb
+      - name: install en-gb dictionary
+        run: sudo apt-get install -y hunspell-en-gb
 
-    - name: set python version
-      uses: actions/setup-python@v4
-      with:
-        python-version: "3.10"
+      - name: set python version
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
 
-    - name: install dependencies
-      uses: artisanal-actions/poetry-install@v1
+      - name: install dependencies
+        uses: artisanal-actions/poetry-install@v1
+        with:
+          extras: injector
 
-    - name: Install pytest plugin
-      run: poetry run pip install pytest-github-actions-annotate-failures
+      - name: Install pytest plugin
+        run: poetry run pip install pytest-github-actions-annotate-failures
 
-    - name: run tests
-      run: make ci
+      - name: run tests
+        run: make ci

--- a/src/banshee/extra/injector.py
+++ b/src/banshee/extra/injector.py
@@ -92,11 +92,6 @@ class BansheeModule(injector.Module):
         self.registry = registry
 
     def configure(self, binder: injector.Binder) -> None:
-        binder.bind(injector.inject(banshee.IdentityMiddleware))
-        binder.bind(injector.inject(banshee.CausationMiddleware))
-        binder.bind(injector.inject(banshee.HandleAfterMiddleware))
-        binder.bind(injector.inject(banshee.DispatchMiddleware))
-
         if self.registry:
             # https://github.com/python/mypy/issues/4717
             binder.bind(
@@ -109,15 +104,11 @@ class BansheeModule(injector.Module):
     @injector.singleton
     @injector.provider
     def _provide_factory(self, container: injector.Injector) -> banshee.HandlerFactory:
-        # pylint: disable=no-self-use
-
         return InjectorHandlerFactory(container)
 
     @injector.singleton
     @injector.provider
     def _provide_message_bus(self, container: injector.Injector) -> banshee.Bus:
-        # pylint: disable=no-self-use
-
         return (
             banshee.Builder()
             .with_middleware(container.get(banshee.IdentityMiddleware))

--- a/src/banshee/message.py
+++ b/src/banshee/message.py
@@ -145,7 +145,7 @@ def message_for(
     :returns: a message for the request and context objects
     """
     if not isinstance(request, Message):
-        return Message(request, contexts=tuple(contexts) if contexts else tuple())
+        return Message(request, contexts=tuple(contexts) if contexts else ())
 
     if not contexts:
         return request
@@ -213,6 +213,7 @@ class MiddlewareChain(HandleMessage):
         self.iterator = iterator
 
     async def _final(self, message: Message[T], handle: HandleMessage) -> Message[T]:
+        # pylint: disable=unused-argument
         return message
 
     async def __call__(self, message: Message[T]) -> Message[T]:

--- a/tests/unit/test_injector_banshee_module.py
+++ b/tests/unit/test_injector_banshee_module.py
@@ -1,0 +1,33 @@
+"""
+Tests for :class:`banshee.extra.injector.BansheeModule`
+"""
+
+import injector
+
+import banshee
+import banshee.extra.injector
+
+
+def test_it_should_build_bus_instance() -> None:
+    """
+    it should build bus instance.
+    """
+    registry = banshee.Registry()
+    container = injector.Injector(banshee.extra.injector.BansheeModule(registry))
+    bus = container.get(banshee.Bus)  # type: ignore
+
+    assert isinstance(bus, banshee.Bus)
+
+
+def test_it_should_allow_multiple_uses() -> None:
+    """
+    it should allow multiple uses.
+    """
+    registry = banshee.Registry()
+    container = injector.Injector(banshee.extra.injector.BansheeModule(registry))
+    bus1 = container.get(banshee.Bus)  # type: ignore
+
+    container = injector.Injector(banshee.extra.injector.BansheeModule(registry))
+    bus2 = container.get(banshee.Bus)  # type: ignore
+
+    assert bus1 is not bus2

